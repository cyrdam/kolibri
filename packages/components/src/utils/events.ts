// TODO: Should be synchronized with enums/events.ts
export type KoliBriEventType =
	| 'blur'
	| 'change'
	| 'changePage'
	| 'changePageSize'
	| 'click'
	| 'close'
	| 'focus'
	| 'input'
	| 'mousedown'
	| 'reset'
	| 'selection-change' // todo remove
	| 'selectionChange'
	| 'sort'
	| 'submit'
	| 'toggle';

export function stopPropagation(event: Event): void {
	event.stopImmediatePropagation();
	event.stopPropagation();
}

function createKoliBriEvent<T>(type: KoliBriEventType, detail: T): CustomEvent {
	const event = new CustomEvent(`kol-${type}`, {
		bubbles: true,
		cancelable: true,
		composed: true,
		detail: detail,
	});
	return event;
}

function dispatchKoliBriEvent<T>(target: EventTarget, type: KoliBriEventType, detail?: T): boolean {
	const dispatch = target.dispatchEvent(createKoliBriEvent(type, detail));
	return dispatch;
}

/**
 * @deprecated
 */
export function tryToDispatchKoliBriEvent<T>(type: KoliBriEventType, target?: EventTarget, detail?: T): void {
	target && dispatchKoliBriEvent(target, type, detail);
}

export function dispatchDomEvent(target: HTMLElement, type: KoliBriEventType) {
	target.dispatchEvent(new Event(type, { bubbles: true, composed: true }));
}

export function preventDefaultAndStopPropagation(event: Event) {
	event.preventDefault();
	stopPropagation(event);
}
