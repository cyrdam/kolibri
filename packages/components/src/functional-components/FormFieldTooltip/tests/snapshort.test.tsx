import { h } from '@stencil/core';
import FormFieldTooltipFc from '../FormFieldTooltip';
import { renderFunctionalComponentToSpecPage } from '../../../utils/testing';

describe('FormFieldTooltipFc', () => {
	it('should render correctly', async () => {
		const label = 'Test Label';
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id="test-id" label={label} />);
		expect(page.root).toMatchSnapshot();
		expect(page.root?.getAttribute('_id')).toBe(null);
		expect(page.root?.getAttribute('_label')).toBe(label);
	});

	it('should set the correct class names', async () => {
		const classNames = 'custom-class';
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id="test-id" label="Test Label" class={classNames} />);

		expect(page.root).toMatchSnapshot();
		expect(page.root?.className).toContain('input-tooltip');
		expect(page.root?.className).toContain(classNames);
	});

	it('should set aria-hidden attribute', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id="test-id" label="Test Label" />);
		expect(page.root?.getAttribute('aria-hidden')).toBe('true');
	});

	it('should set the correct badgeText', async () => {
		const badgeText = 'Badge Text';
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id="test-id" label="Test Label" badgeText={badgeText} />);

		expect(page.root).toMatchSnapshot();
		expect(page.root?.getAttribute('_badgeText')).toBe(badgeText);
	});

	it('should set the correct align property', async () => {
		const align = 'left';
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id="test-id" label="Test Label" align={align} />);

		expect(page.root).toMatchSnapshot();
		expect(page.root?.getAttribute('_align')).toBe(align);
	});

	it('should set the correct id and label', async () => {
		const id = 'test-id';
		const label = 'Test Label';
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id={id} label={label} />);

		expect(page.root).toMatchSnapshot();
		expect(page.root?.getAttribute('_id')).toBe(null);
		expect(page.root?.getAttribute('_label')).toBe(label);
	});

	it('should set the correct id when hideLabel is true', async () => {
		const id = 'test-id';
		const label = 'Test Label';
		const page = await renderFunctionalComponentToSpecPage(() => <FormFieldTooltipFc id={id} label={label} hideLabel={true} />);

		expect(page.root).toMatchSnapshot();
		expect(page.root?.getAttribute('_id')).toBe(`${id}-label`);
		expect(page.root?.getAttribute('_label')).toBe(label);
	});
});
