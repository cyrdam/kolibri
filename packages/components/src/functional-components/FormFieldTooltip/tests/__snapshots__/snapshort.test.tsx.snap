// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldTooltipFc should render correctly 1`] = `
<kol-tooltip-wc
  _label="Test Label"
  aria-hidden="true"
  class="kol-form-field__tooltip"
/>
`;

exports[`FormFieldTooltipFc should set the correct align property 1`] = `
<kol-tooltip-wc
  _align="left"
  _label="Test Label"
  aria-hidden="true"
  class="kol-form-field__tooltip"
/>
`;

exports[`FormFieldTooltipFc should set the correct badgeText 1`] = `
<kol-tooltip-wc
  _badgetext="Badge Text"
  _label="Test Label"
  aria-hidden="true"
  class="kol-form-field__tooltip"
/>
`;

exports[`FormFieldTooltipFc should set the correct class names 1`] = `
<kol-tooltip-wc
  _label="Test Label"
  aria-hidden="true"
  class="kol-form-field__tooltip custom-class"
/>
`;

exports[`FormFieldTooltipFc should set the correct id and label 1`] = `
<kol-tooltip-wc
  _label="Test Label"
  aria-hidden="true"
  class="kol-form-field__tooltip"
/>
`;

exports[`FormFieldTooltipFc should set the correct id when hideLabel is true 1`] = `
<kol-tooltip-wc
  _id="test-id-label"
  _label="Test Label"
  aria-hidden="true"
  class="kol-form-field__tooltip"
/>
`;
